### DISCUSSION TITLE
custom clipping with backgroundImage

### DISCUSSION THREAD

**Discussion by metiskot** [fabricjs/fabric.js#7747] [Q&A]
Hello!

I use the setBackgroundImage and globalCompositeOperation property to crop the canvas properly. 

But when saving via toDataURL, I need to remove only the background, but so that all the objects that were cropped by globalCompositeOperation are preserved. 

Now, if I delete or make the background transparent, then these actions also affect objects. How can this be circumvented?

**Comment by ShaMan123** [comment]
I don't understand how you do that but you should be able to leverage the canvas' clipPath property
